lib/Db/TagMapper.php

```diff
@@ -156,9 +156,13 @@ public function getAllTagsForMessages(array $messages): array {
 		}
 		$result = [];
 		foreach ($queryResult as $qr) {
-			$result[] = $qr['imap_message_id'];
-			$result[$qr['imap_message_id']][] = $this->getTag((int)$qr['tag_id']);
-		};
+			try {
+				$result[] = $qr['imap_message_id'];
```

Comment:
What exactly is the purpose of this line? IIRC, the returned result is expected to be a map of `message_id => tag[]` while this line is also creating an array of message ids.

Example output for `$result` (message with one tag):
```php
array (
  0 => '<20210308112210.Horde.djiH_5Ms4hawgi2WUoVHzTD@localhost>',
  '<20210308112210.Horde.djiH_5Ms4hawgi2WUoVHzTD@localhost>' => 
  array (
    0 => 
    OCA\Mail\Db\Tag::__set_state(array(
       'userId' => 'admin',
       'displayName' => 'Important',
       'imapLabel' => '$label1',
       'color' => '#FF0000',
       'isDefaultTag' => true,
       'id' => 1,
       '_updatedFields' => 
      array (
      ),
       '_fieldTypes' => 
      array (
        'id' => 'integer',
        'isDefaultTag' => 'boolean',
      ),
    )),
  ),
)
```

Reply:
Yep, it was really not well written. will be removed with #4822